SpinCore_pp/pulse_length_conv.py

```diff
@@ -1,79 +1,124 @@
+                    contains the following keys.  It's crucial that these get used in the pulse sequence.
+
+                    :amplitude: float
+                    :deblank_us: float
+
     Returns
     =======
     retval: float
-            The pulse length you tell spincore in order to get the desired actual.
+            The pulse length you tell spincore in order to get the desired β.
     """
-    # {{{ list of programmed p90, actual p90 and actual 180 - used in
-    # generating the calibrated fit
-    # list of the programmed pulse length, the actual p90 length
-    # and the actual p180 length based on the programmed p90
-    datapoints = [
-        (1, 2.25869, 5.75412),
-        (2, 5.78065, 16.6168),
-        (3, 10.4132, 36.0229),
-        (3.5, 13.2053, 49.4104),
-        (4, 16.9808, 63.7255),
-        (4.5, 20.4275, 77.5474),
-        (5, 24.8538, 91.5177),
-        (5.5, 30.0159, 106.302),
-        (6, 35.6903, 119.849),
-        (6.5, 42.2129, 134.328),
-        (7, 48.9577, 149.029),
-        (7.5, 55.977, 162.774),
-        (8, 63.0445, 176.917),
-        (8.5, 69.881, 190.167),
-        (9, 76.9727, 203.486),
-        (9.5, 84.5149, 217.965),
-        (10, 91.7002, 232.042),
-        (10.5, 98.6599, 245.213),
-        (11, 105.327, 258.004),
-        (11.5, 112.985, 272.776),
-        (12, 120.264, 286.884),
-        (12.5, 126.251, 299.009),
-    ]
-    # neat JF trick for organizing these data points
-    prog90, act90, act180 = map(array, zip(*datapoints))
-    # }}}
-    # {{{ prepare data into arrays for interpolation
-    # gather programmed pulse lengths in array
-    plen_prog = r_[0, prog90, 2 * prog90]
-    # assume the longest pulse is about the correct length
-    # and again gather into an array
-    plen_actual = r_[0, act90, act180] * 2 * prog90[-1] / act180[-1]
+    assert isinstance(
+        settings, spc.config_parser_fn.configuration
+    ), "You need to pass your configuration dict so I know what the amplitude and deblank time are"
 
-    # }}}
-    def zonefit(desired_actual):
-        if desired_actual < 25:
-            mask = np.ones_like(plen_prog, dtype=bool)
-        else:
-            mask = plen_prog > 10
-        calibration_data = nddata(plen_prog[mask], [-1], ["plen"]).setaxis(
-            "plen", plen_actual[mask]
+    if np.isscalar(desired_beta):
+        assert (
+            desired_beta < 1000e-6
+        ), "You asked for a desired beta of over 1,000 μs√W.  This is not the beta value you are looking for!!!"
+    else:
+        assert (
+            desired_beta[-1] < 1000e-6
+        ), "You asked for a desired beta of over 1,000 μs√W.  This is not the beta value you are looking for!!!"
+    assert (
+        settings["deblank_us"] == 50
+    ), "currently only calibrated for deblank_us = 50, so you almost definitely want to set that value in your active.ini"
+    if settings["amplitude"] == 1.0:
+        linear_threshold = 100e-6
+        c_nonlinear = [
```

Comment:
these should all be `r_`
which means at the top, you need `from numpy import r_`

Reply:
(`r_` gives an array -- otherwise it's a list)